src/fsd/4-entities/upgrade/upgrades.service.ts

```
@@ -23,6 +26,178 @@
+        // Finally, perform a BFS to fill in all expansions that
+        // have more than one additional layer.
+        //
+        // As of 2025-01-01, it takes three passes (one of which is above) to fully expand all recipe data.
+        let passes: number = 0;
+        const kNumExpectedPasses = 2;
+        for (let moreToExpand: boolean = true; moreToExpand; ) {
+            ++passes;
+            moreToExpand = false;
+            Object.entries(UpgradesService.craftedUpgradesData).forEach(data => {
+                const material: ICraftedUpgrade = data[1];
+                const expandedRecipe: IRecipeExpandedUpgrade | null = this.expandRecipe(material.id, result);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)